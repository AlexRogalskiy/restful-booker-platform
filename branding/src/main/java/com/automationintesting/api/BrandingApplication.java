package com.automationintesting.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan()
public class BrandingApplication {

    public static void main(String[] args) {
        SpringApplication.run(BrandingApplication.class, args);
    }

}
