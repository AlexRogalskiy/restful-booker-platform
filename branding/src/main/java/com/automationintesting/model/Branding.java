package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class Branding {

    @JsonProperty
    private String name;
    @JsonProperty
    private Map map;
    @JsonProperty
    private String logoUrl;
    @JsonProperty
    private String description;
    @JsonProperty
    private Contact contact;

    public Branding() {}

    public Branding(String name, Map map, String logoUrl, String description, Contact contact) {
        this.name = name;
        this.map = map;
        this.logoUrl = logoUrl;
        this.description = description;
        this.contact = contact;
    }

    public Branding(ResultSet result) throws SQLException {
        this.name = result.getString("name");
        this.map = new Map(result);
        this.logoUrl = result.getString("logo_url");
        this.description = result.getString("description");
        this.contact = new Contact(result);
    }

    public String getName() {
        return name;
    }

    public Map getMap() {
        return map;
    }

    public String getLogoUrl() {
        return logoUrl;
    }

    public String getDescription() {
        return description;
    }

    public Contact getContact() {
        return contact;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setMap(Map map) {
        this.map = map;
    }

    public void setLogoUrl(String logoUrl) {
        this.logoUrl = logoUrl;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setContact(Contact contact) {
        this.contact = contact;
    }

    @Override
    public String toString() {
        return "Branding{" +
                "name='" + name + '\'' +
                ", map=" + map +
                ", logoUrl='" + logoUrl + '\'' +
                ", description='" + description + '\'' +
                ", contact=" + contact +
                '}';
    }
}
