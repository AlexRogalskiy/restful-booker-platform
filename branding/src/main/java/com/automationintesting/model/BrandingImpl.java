package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class BrandingImpl implements Branding {

    @JsonProperty
    private Map map;
    @JsonProperty
    private Logo logo;
    @JsonProperty
    private String description;
    @JsonProperty
    private Contact contact;

    public BrandingImpl(Map map, Logo logo, String description, Contact contact) {
        this.map = map;
        this.logo = logo;
        this.description = description;
        this.contact = contact;
    }

    public BrandingImpl(ResultSet result) throws SQLException {
        this.map = new MapImpl();
        this.logo = new LogoImpl();
        this.description = result.getString("description");
        this.contact = new ContactImpl();
    }

    @Override
    public Map getMap() {
        return map;
    }

    @Override
    public Logo getLogo() {
        return logo;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public Contact getContact() {
        return contact;
    }

}
