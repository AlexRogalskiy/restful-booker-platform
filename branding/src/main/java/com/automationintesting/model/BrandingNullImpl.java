package com.automationintesting.model;

public class BrandingNullImpl implements Branding {

    @Override
    public Map getMap() {
        return new MapNullImpl();
    }

    @Override
    public Logo getLogo() {
        return new LogoNullImpl();
    }

    @Override
    public String getDescription() {
        return "";
    }

    @Override
    public Contact getContact() {
        return new ContactNullImpl();
    }

}
