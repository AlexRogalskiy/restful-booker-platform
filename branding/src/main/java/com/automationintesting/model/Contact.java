package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class Contact {

    @JsonProperty
    private String name;
    @JsonProperty
    private String address;
    @JsonProperty
    private String phone;
    @JsonProperty
    private String email;

    public Contact(String name, String address, String phone, String email) {
        this.name = name;
        this.address = address;
        this.phone = phone;
        this.email = email;
    }

    public Contact(ResultSet result) throws SQLException {
        this.name = result.getString("contact_name");
        this.address = result.getString("address");
        this.phone = result.getString("phone");
        this.email = result.getString("email");
    }

    public Contact() {}

    public String getName() {
        return name;
    }

    public String getAddress() {
        return address;
    }

    public String getPhone() {
        return phone;
    }

    public String getEmail() {
        return email;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}
