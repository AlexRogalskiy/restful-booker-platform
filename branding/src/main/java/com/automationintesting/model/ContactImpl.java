package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ContactImpl implements Contact {

    @JsonProperty
    private String name;
    @JsonProperty
    private String address;
    @JsonProperty
    private String phone;
    @JsonProperty
    private String email;

    public ContactImpl() {
        this.name = "Shady Meadows B&B";
        this.address = "The Old Farmhouse, Shady Street, Newfordburyshire, NE1 410S";
        this.phone = "0123456789";
        this.email = "fake@fakeemail.com";
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getAddress() {
        return address;
    }

    @Override
    public String getPhone() {
        return phone;
    }

    @Override
    public String getEmail() {
        return email;
    }

}
