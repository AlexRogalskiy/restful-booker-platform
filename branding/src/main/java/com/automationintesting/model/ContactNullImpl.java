package com.automationintesting.model;

public class ContactNullImpl implements Contact {

    @Override
    public String getName() {
        return "";
    }

    @Override
    public String getAddress() {
        return "";
    }

    @Override
    public String getPhone() {
        return "";
    }

    @Override
    public String getEmail() {
        return "";
    }

}
