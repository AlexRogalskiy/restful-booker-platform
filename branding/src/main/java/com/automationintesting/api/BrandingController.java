package com.automationintesting.api;

import com.automationintesting.db.BrandingDB;
import com.automationintesting.model.Branding;
import com.automationintesting.requests.AuthRequests;
import org.springframework.context.annotation.Bean;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.sql.SQLException;

@RestController
public class BrandingController {

    private BrandingDB brandingDB;
    private AuthRequests authRequest;

    @Bean
    public WebMvcConfigurer configurer() {

        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedMethods("GET", "POST", "DELETE", "PUT")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    public BrandingController() throws SQLException {
        this.brandingDB = new BrandingDB();
        this.authRequest = new AuthRequests();
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public ResponseEntity<Branding> getBranding() throws SQLException {
        Branding branding = brandingDB.queryBranding();

        return ResponseEntity.ok(branding);
    }

    @RequestMapping(value = "/", method = RequestMethod.PUT)
    public ResponseEntity<?> createBooking(@RequestBody Branding branding, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequest.postCheckAuth(token)){
            Branding updatedBranding = brandingDB.update(branding);

            return ResponseEntity.ok(updatedBranding);
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

}
