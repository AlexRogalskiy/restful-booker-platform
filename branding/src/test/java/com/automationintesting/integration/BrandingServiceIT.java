package com.automationintesting.integration;

import com.automationintesting.api.BrandingApplication;
import com.automationintesting.model.*;
import com.github.tomakehurst.wiremock.common.ConsoleNotifier;
import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
import com.github.tomakehurst.wiremock.junit.WireMockRule;
import io.restassured.http.ContentType;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static com.github.tomakehurst.wiremock.client.WireMock.*;
import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static io.restassured.RestAssured.given;
import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, classes = BrandingApplication.class)
@ActiveProfiles("dev")
public class BrandingServiceIT {

    @Rule
    public WireMockRule wireMockRule = new WireMockRule(WireMockConfiguration.options().notifier(new ConsoleNotifier(true)).port(3004));

    @Before
    public void setupWiremock(){
        stubFor(post("/auth/validate")
                .withRequestBody(equalToJson("{ \"token\": \"abc123\" }"))
                .willReturn(aResponse().withStatus(200)));
    }

    @Test
    public void returnsBrandingData() {
        Response brandingResponse =  given()
                .when()
                .get("http://localhost:3002/branding/");

        Approvals.verify(brandingResponse.getBody().prettyPrint());
    }

    @Test
    public void updateBrandingData() {
        Branding brandingPayload = new Branding(
                "Updated hotel name",
                new Map(50.0, 50.0),
                "link/to/logo",
                "Description update",
                new Contact("Update name", "Update address", "9999999999", "update@email.com")
        );

        Response brandingPutResponse = given()
                .cookie("token", "abc123")
                .contentType(ContentType.JSON)
                .body(brandingPayload)
                .when()
                .put("http://localhost:3002/branding/");

        Approvals.verify(brandingPutResponse.body().prettyPrint());
    }

    @Test
    public void testPutValidation() {
        Branding brandingPayload = new Branding.BrandingBuilder()
                .build();

        Response response = given()
                .contentType(ContentType.JSON)
                .body(brandingPayload)
                .when()
                .put("http://localhost:3002/branding/");

        assertThat(response.statusCode(), is(400));
    }

}
