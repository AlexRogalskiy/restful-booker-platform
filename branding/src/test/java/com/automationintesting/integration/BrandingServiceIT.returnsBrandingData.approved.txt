{
    "name": "Sauce Labs Retreat",
    "map": {
        "latitude": 37.7871917,
        "longitude": -122.4005403
    },
    "logoUrl": "https://automationintesting.com/images/supporters/saucelabs/020101_LOGO_Sauce-Labs_Vert_Red-Grey_RGB.png",
    "description": "Welcome to the Sauce Labs Retreat, the perfect space for continuous testing. Founded in 2008 and set in the glorious cross browser foothills of Foxfire and Pareo in the beautiful cloud, here you can take part in a wide range of activities including acting lessons for devices, painting for visual comparison and language lessons for Selenium, Appium, Espresso and more. So join us for your continuous testing solutions, stay for the hot sauce.",
    "contact": {
        "name": "Sauce Labs - USA (HQ)",
        "address": "116 New Montgomery St, 3rd Floor San Francisco, CA 94105 USA",
        "phone": "012345678901",
        "email": "https://saucelabs.com/sales"
    }
}
